### PR TITLE
Cosmetic improvements

### PR DESCRIPTION
**Pull request by Vilin97**
A purely cosmetic PR: added docstrings, `Base.show` functions, exporting more so `DiffEqJump.` does not have to be used in defining hopping constants.

### CODE REVIEW

File: src/spatial/hop_rates.jl [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    hopping_constants::Matrix{F} 
+
+    "rates[i,j] is total hopping rate of species i at site j"
+    rates::Matrix{F}
+
+    "sum_rates[j] is the sum of hopping rates at site j"
+    sum_rates::Vector{F}
+end
+
+function Base.show(io::IO, hop_rates::HopRatesUnifNbr)
```
Comment (ChrisRackauckas) [inline src/spatial/hop_rates.jl:41]: This is the wrong show dispatch. This probably should be for the text mimetype. 2-argument should is supposed to be round-trippable.

File: src/spatial/hop_rates.jl [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    hop_const_cumulative_sums::Matrix{Vector{F}} 
+
+    "rates[s,i] is the total hopping rate of species s at site i"
+    rates::Matrix{F}
+
+    "sum_rates[i] is the total hopping rate out of site i"
+    sum_rates::Vector{F}
+end
+
+function Base.show(io::IO, hop_rates::HopRatesGeneral)
```
Comment (ChrisRackauckas) [inline src/spatial/hop_rates.jl:123]: same here.

File: src/spatial/reaction_rates.jl [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+# helper functions
+function set_rx_rate_at_site!(rx_rates::RxRates, site, rx, rate)
+    @inbounds old_rate = rx_rates.rates[rx, site]
+    @inbounds rx_rates.rates[rx, site] = rate
+    @inbounds rx_rates.sum_rates[site] += rate - old_rate
+    old_rate
+end
+
+function Base.show(io::IO, rx_rates::RxRates)
```
Comment (ChrisRackauckas) [inline src/spatial/reaction_rates.jl:79]: and here

File: src/spatial/topology.jl [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    nums_neighbors = [count(x -> x+CI[site] in CI, offsets) for site in 1:prod(dims)]
+    CartesianGridIter(dims, nums_neighbors, CI, LI, offsets)
+end
+CartesianGridIter(dims) = CartesianGridIter(Tuple(dims))
+CartesianGridIter(dimension, linear_size::Int) = CartesianGridIter([linear_size for i in 1:dimension])
+function rand_nbr(grid::CartesianGridIter, site::Int)
+    nth_nbr(grid, site, rand(1:num_neighbors(grid,site)))
+end
+
+function Base.show(io::IO, grid::Union{CartesianGridRej, CartesianGridIter})
```
Comment (ChrisRackauckas) [inline src/spatial/topology.jl:124]: and here.

File: src/spatial/hop_rates.jl (hunk 1 above)
Comment (Vilin97) [reply src/spatial/hop_rates.jl]: Thanks. Fixed

**Review by isaacsas — APPROVED** [review]

**Comment by isaacsas** [comment]
@Vilin97 is this good to go on your end?

**Pull request merged**